underscore.js

```diff
@@ -685,7 +685,8 @@
     var low = 0, high = array.length;
     while (low < high) {
       var mid = Math.floor((low + high) / 2);
-      if (iteratee(array[mid]) < value) low = mid + 1; else high = mid;
+      var cur = iteratee(array[mid]);
+      if ( cur < value || (value === void 0 && cur <= Infinity) || (_.isNaN(value) && (cur <= Infinity || cur === void 0))) low = mid + 1; else high = mid;
```

Comment:
I've been pretty busy the past couple weeks (exam time) but I'm playing with this logic now. I've almost got it passing (1 failing test) with this

```js
if ( cur < value || (cur <= Infinity && !(value <= value)) ) low = mid + 1;
else high = mid;
```